test/Conversion/SCFToCalyx/convert_controlflow.mlir

```diff
@@ -0,0 +1,130 @@
+// RUN: circt-opt %s --lower-scf-to-calyx -split-input-file | FileCheck %s
+
+// CHECK:      module  {
+// CHECK-NEXT:   calyx.program "main"  {
+// CHECK-NEXT:     calyx.component @main(%in0: i32, %in1: i32, %clk: i1 {clk}, %reset: i1 {reset}, %go: i1 {go}) -> (%out0: i32, %done: i1 {done}) {
+// CHECK-NEXT:       %true = hw.constant true
+// CHECK-NEXT:       %std_slt_0.left, %std_slt_0.right, %std_slt_0.out = calyx.std_slt "std_slt_0" : i32, i32, i1
+// CHECK-NEXT:       %ret_arg0_reg.in, %ret_arg0_reg.write_en, %ret_arg0_reg.clk, %ret_arg0_reg.reset, %ret_arg0_reg.out, %ret_arg0_reg.done = calyx.register "ret_arg0_reg" : i32, i1, i1, i1, i32, i1
+// CHECK-NEXT:       %bb3_arg0_reg.in, %bb3_arg0_reg.write_en, %bb3_arg0_reg.clk, %bb3_arg0_reg.reset, %bb3_arg0_reg.out, %bb3_arg0_reg.done = calyx.register "bb3_arg0_reg" : i32, i1, i1, i1, i32, i1
+// CHECK-NEXT:       calyx.wires  {
+// CHECK-NEXT:         calyx.assign %out0 = %ret_arg0_reg.out : i32
+// CHECK-NEXT:         calyx.comb_group @bb0_0  {
+// CHECK-NEXT:           calyx.assign %std_slt_0.left = %in0 : i32
+// CHECK-NEXT:           calyx.assign %std_slt_0.right = %in1 : i32
+// CHECK-NEXT:         }
+// CHECK-NEXT:         calyx.group @bb1_to_bb3  {
+// CHECK-NEXT:           calyx.assign %bb3_arg0_reg.in = %in0 : i32
+// CHECK-NEXT:           calyx.assign %bb3_arg0_reg.write_en = %true : i1
+// CHECK-NEXT:           calyx.group_done %bb3_arg0_reg.done : i1
+// CHECK-NEXT:         }
+// CHECK-NEXT:         calyx.group @bb2_to_bb3  {
+// CHECK-NEXT:           calyx.assign %bb3_arg0_reg.in = %in1 : i32
+// CHECK-NEXT:           calyx.assign %bb3_arg0_reg.write_en = %true : i1
+// CHECK-NEXT:           calyx.group_done %bb3_arg0_reg.done : i1
+// CHECK-NEXT:         }
+// CHECK-NEXT:         calyx.group @ret_assign_0  {
+// CHECK-NEXT:           calyx.assign %ret_arg0_reg.in = %bb3_arg0_reg.out : i32
+// CHECK-NEXT:           calyx.assign %ret_arg0_reg.write_en = %true : i1
+// CHECK-NEXT:           calyx.group_done %ret_arg0_reg.done : i1
+// CHECK-NEXT:         }
+// CHECK-NEXT:       }
+// CHECK-NEXT:       calyx.control  {
+// CHECK-NEXT:         calyx.seq  {
+// CHECK-NEXT:           calyx.if %std_slt_0.out with @bb0_0  {
+// CHECK-NEXT:             calyx.seq  {
+// CHECK-NEXT:               calyx.seq  {
+// CHECK-NEXT:               }
+// CHECK-NEXT:               calyx.seq  {
+// CHECK-NEXT:                 calyx.enable @bb1_to_bb3 {compiledGroups = []}
```

Comment:
If you remove the attr-dict here from the calyx.enable, will it not work? We should clean this up and make it optional in a future PR. Not including it here will make it easier.

Reply:
Looks like `CHECK-NEXT` matches on the entire line, so unfortunately removing the latter part of the check makes the test fail. And i agree, having the `compiledGroups` print for the `calyx.enable` by default should be revised.

Reply:
Ok, opened issue in #1844 to address this at some point.